discord/message.py

```diff
@@ -237,6 +237,42 @@ def cached_message(self):
         """Optional[:class:`Message`]: The cached message, if found in the internal message cache."""
         return self._state._get_message(self.message_id)
 
+    async def fetch_message(self):
+        """|coro|
+
+        Retrieves the :class:`Message` from the destination.
+
+        This can only be used by bot accounts.
+
+        .. versionadded:: 2.0
+
+        Parameters
+        ------------
+        id: :class:`int`
+            The message ID to look for.
+
+        Raises
+        --------
+        ~discord.NotFound
+            The message was not found.
+        ~discord.ChannelNotFound
+            The message's channel was not found.
+        ~discord.Forbidden
+            You do not have the permissions required to get a message.
+        ~discord.HTTPException
+            Retrieving the message failed.
+
+        Returns
+        --------
+        :class:`Message`
+            The message asked for.
+        """
+
+        channel = self._state.get_channel(self.channel_id)
```

Comment:
This could be more efficient but `ConnectionState` doesn't publicly expose any methods that make this simple. I considered adding a `guild_id` parameter to `ConnectionState.get_channel` but iterating over every guild should still be fast as there are no API requests

Reply:
Personally I avoid iterating over every guild like the plague as it's way too easy to not notice it does this and block the bot for a noticeable period when you have a lot of guilds and call it relatively often

Reply:
I could either implement a `guild_id` parameter or just fetch the channel (but have to do possibly an unnecessary API request)
@Rapptz which would you prefer?

Reply:
There's already a `MessageReference.guild_id`

Reply:
I meant add `guild_id` to `ConnectionState.get_channel`